### PR TITLE
[fix](routine-load) fix get Kafka offset timeout may too long

### PR DESCRIPTION
**Pull request by sollhui**
## Proposed changes

```
for (int32_t partition_id : partition_ids) {
        int64_t low = 0;
        int64_t high = 0;
        RdKafka::ErrorCode err =
                _k_consumer->query_watermark_offsets(_topic, partition_id, &low, &high, 5);
        if (err != RdKafka::ERR_NO_ERROR) {
            std::stringstream ss;
            ss << "failed to get latest offset for partition: " << partition_id
               << ", err: " << RdKafka::err2str(err);
            LOG(WARNING) << ss.str();
            return Status::InternalError(ss.str());
        }
```

fe getInfo timeout is 5s, but in be, the timeout is 5 * partition num.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by HHoflittlefish777** [comment]
run buildall

**Comment by dataroaring** [comment]
run buildall

**Comment by HHoflittlefish777** [comment]
run buildall

**Comment by HHoflittlefish777** [comment]
run buildall

**Comment by HHoflittlefish777** [comment]
run buildall

**Comment by HHoflittlefish777** [comment]
run p0

**Comment by HHoflittlefish777** [comment]
run buildall

**Comment by HHoflittlefish777** [comment]
run buildall

**Comment by HHoflittlefish777** [comment]
run buildall

**Comment by HHoflittlefish777** [comment]
run buildall

**Review by dataroaring — DISMISSED** [review]
LGTM

**Comment by HHoflittlefish777** [comment]
run buildall

**Review by dataroaring — DISMISSED** [review]
LGTM

File: be/src/runtime/routine_load/data_consumer.cpp [hunk 1]
```diff
@@ -384,13 +384,21 @@ Status KafkaDataConsumer::get_offsets_for_times(const std::vector<PIntegerPair>&
...
 Status KafkaDataConsumer::get_latest_offsets_for_partitions(
-        const std::vector<int32_t>& partition_ids, std::vector<PIntegerPair>* offsets) {
+        const std::vector<int32_t>& partition_ids, std::vector<PIntegerPair>* offsets,
+        int timeout) {
+    MonotonicStopWatch watch;
+    watch.start();
     for (int32_t partition_id : partition_ids) {
         int64_t low = 0;
         int64_t high = 0;
+        auto timeout_ms = timeout - watch.elapsed_time() / 1000 / 1000;
```
Comment (liaoxin01) [inline be/src/runtime/routine_load/data_consumer.cpp:394]: watch.elapsed_time() return uint64_t, implicit type conversion may lead to unexpected issues.

**Review by liaoxin01 — CHANGES_REQUESTED** [review]

**Comment by HHoflittlefish777** [comment]
run buildall

**Comment by HHoflittlefish777** [comment]
run buildall

**Review by liaoxin01 — APPROVED** [review]
LGTM

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**